### PR TITLE
Refactor for DataValue

### PR DESCRIPTION
**Pull request by victlu**
Refactor out `value` portion of IDataPoint to a IDataValue.

### CODE REVIEW

File: src/OpenTelemetry/Metrics/DataPoint/DataValue.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+namespace OpenTelemetry.Metrics
+{
+    public readonly struct DataValue : IDataValue
+    {
+        private readonly IDataValue value;
+
+        internal DataValue(int value)
+        {
+            // Promote to long
+            this.value = new DataValue<int>(value);
```
Comment (CodeBlanch) [inline src/OpenTelemetry/Metrics/DataPoint/DataValue.cs:28]: Just double-checking, did you mean to do `DataValue<long>`?

Comment (victlu) [reply src/OpenTelemetry/Metrics/DataPoint/DataValue.cs:28]: Yes. My typo.

**Review by pirgeo — APPROVED** [review]
LGTM, I just have one question: In the proto and the spec, `Attribute` or `Label` is used as the name for Key-Value pairs, whereas here, `Tag` is used - I might be comparing apples to oranges here, so if I do, please let me know, but would it make sense to keep the naming consistent everywhere as much as possible? Or is there some guidance on when to use what that that I simply missed?

**Comment by victlu** [comment]
@pirgeo You are correct. We need to rename 'tag' to 'attribute'. Just haven't gotten to it yet.

**Review by cijothomas — APPROVED** [review]

**Pull request merged**